source/tutorials/datapacks.md

```diff
@@ -0,0 +1,338 @@
+|                                | <code>"quantity"</code>               | integer    | A number of items to be consumed. Pairs with "item", defaults to 1. |
+|                                | <code>"building"</code>               | string     | A specific building that must be present in the colony. Normal "building" requirements combine the levels from multiple of the same building type. Optionally pairs with "level", otherwise defaults to level 1. |
+|                                | <code>"mandatory-building"</code>     | string     | A specific building that must be present in the colony, with at least one occurence at least at the specifed level. Optionally pairs with "level", otherwise defaults to level 1. |
+|                                | <code>"alternate-building"</code>     | string     | A specific building to be added to the list of alternate-buildings. At least one of the alternate-buildings must be present in the colony to begin research. Optionally pairs with "level", otherwise defaults to level 1. Acts like "building" if only one <code>alternate-building</code> object is present. |
+|                                | <code>"level"</code>                  | integer    | Specifies a building level. Pairs with <code>building</code>, <code>mandatory-building</code>, and <code>alternate-building</code>, otherwise has no effect. If not present for those requirements, defaults to 1. |
+|                                | <code>"research"</code>               | string     | A specific Research by Resource Location that must be completed to unlock this research. Unlike, <code>parentResearch</code>, can be on another branch, or of any research level. |
+| <code>"effects"</code>         |                                       | Array Of Objects | Contains all of the effects of the research. Can be empty, either for research with no effects or for research that only unlocks crafterrecipes, or contain one or more Objects. |
+|                                | <code>"string": integer</code>        | Name-Value Pair | The Research Effect Identifier as a Resource Location, followed by the level of the effect as an integer. If the level exceeds the maximum level of the effect, defaults to the maximum level. If no matching effect is found, applies a default strength of 6, enough to unlock a building completely. |
+| <code>"remove"</code>          | boolean or string or Array of Strings | Removes a research, regardless of the source. May cause data loss if applied to an existing world. If a string, removes any research matching the Resource Location in that string. If an array of strings, removes all researches matching any of the Resource Locations. If boolean and true, removes Research matching the Resource Location of the JSON itself; this form is discouraged for cases where other data packs may apply, as it may be conflicted by those other data packs. |
+{% comment %}
```

Comment:
Why is this a comment?

Reply:
Right now, `icon` is disabled in the mainline MineColonies code: it's read, but the check texture always is used instead.  It's easy enough to turn back on, but until there's a serious decision made as to whether it'll stick in the long-term -- or if it'll only be used in certain `branch-types` -- I don't want to encourage people to use it.  At the same time, I don't want the information to be missing.